EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -2105,4 +2116,61 @@ void V2::World::updateCountryDetails()
 			}
 		}
 	}
+}
+
+void V2::World::processShatteredHre(const std::optional<std::string>& eu4HreTag)
+{
+	if (!eu4HreTag)
+		return;
+
+	const auto& v2HreTag = countryMapper.getV2Tag(*eu4HreTag);
+	if (!v2HreTag)
+		return;
+
+	decisions["shattered_hre.txt"] = customizeFile("blankMod/output/decisions/shattered_hre.txt", std::regex("\\bHRE\\b"), *v2HreTag);
+	events["shattered_hre.txt"] = customizeFile("blankMod/output/events/shattered_hre.txt", std::regex("\\bHRE\\b"), *v2HreTag);
+
+	const auto& tagAdj = getCountry(*v2HreTag)->getLocalisation().getLocalAdjective();
+	localisations["0_shattered_hre.csv"] = customizeFile("blankMod/output/localisation/0_shattered_hre.csv", std::regex("\\bHoly Roman\\b"), tagAdj);
+}
+
+std::string V2::World::customizeFile(const std::string& filePath,
+	 const std::regex& oldRegex,
+	 const std::string& newString)
+{
+	std::ifstream input(filePath);
+	if (!input.is_open())
+		throw std::runtime_error("Could not open " + filePath + " for reading!");
+	std::stringstream inStream;
+	inStream << input.rdbuf();
+	auto fileString = inStream.str();
+	input.close();
+	return std::regex_replace(fileString, oldRegex, newString);
+}
+
+void V2::World::outputDynamicContent() const
+{
+	for (const auto& [file, decision]: decisions)
+	{
+		std::ofstream out("output/" + theConfiguration.getOutputName() + "/decisions/" + file);
+		if (!out.is_open())
+			throw std::runtime_error("Could not open " + file + " for writing!");
+		out << decision;
+	}
+
+	for (const auto& [file, event]: events)
+	{
+		std::ofstream out("output/" + theConfiguration.getOutputName() + "/events/" + file);
+		if (!out.is_open())
+			throw std::runtime_error("Could not open " + file + " for writing!");
+		out << event;
+	}
+
+	for (const auto& [file, localisation]: localisations)
+	{
+		std::ofstream out("output/" + theConfiguration.getOutputName() + "/localisation/" + file);
+		if (!out.is_open())
+			Log(LogLevel::Debug) << "Could not open " + file + " for writing!";
```

Comment:
throw exception, not debug.

Reply:
> throw exception, not debug.

I was thinking, missing localisation doesn't break anything, so instead throwing exception make a log entry. But maybe Info instead of Debug?

Reply:
if you can't open a file locally and you could previously, someone ran out of space on disk. exception is the correct choice to stop execution asap.